x/profiles/keeper/keeper_relationships.go

```diff
@@ -1,37 +1,46 @@
 package keeper
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 	"github.com/desmos-labs/desmos/x/profiles/types"
 )
 
 // SaveRelationship allows to store the given relationship returning an error if he's already present.
 func (k Keeper) SaveRelationship(ctx sdk.Context, relationship types.Relationship) error {
 	store := ctx.KVStore(k.storeKey)
-	key := types.RelationshipsStoreKey(relationship.Creator)
+	key := types.RelationshipsStoreKey(relationship.Creator, relationship.Subspace, relationship.Recipient)
 
-	relationships := types.MustUnmarshalRelationships(k.cdc, store.Get(key))
-	for _, rel := range relationships {
-		if rel.Equal(relationship) {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest,
-				"relationship already exists with %s", relationship.Recipient)
-		}
+	if store.Has(key) {
+		return fmt.Errorf("the relationship has already existed")
```

Comment:
I think it's better to stick with using `sdkerrors` instead of `fmt.Errorf`: 

```suggestion
		
			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "relationship already exists with %s", relationship.Recipient)
```